src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -298,19 +276,70 @@ private async Task<bool> WriteResultsAsync(TContext rootContext)
                     catch (Exception e)
                     {
                         context?.Dispose();
-                        context = default;
                         RuntimeErrors |= Errors.LogUnhandledEngineException(rootContext, e);
                         ThrowExitApplicationException(context, ExitReason.ExceptionWritingToLogFile, e);
                     }
                 }
             }
 
-            Debug.Assert(_fileContexts.Count == 0);
+            Debug.Assert(_fileContexts.IsEmpty);
             Debug.Assert(_fileContextsCount == currentIndex);
 
             return true;
         }
 
+        private void LogCachingLogger(TContext rootContext, IAnalysisLogger logger, TContext context, bool clone = false)
```

Comment:
>LogCachingLogger [](http://example.com/codeflow?start=21&length=16)

This is the method that will either clone a previous cached result or just save it.

The clone will be used when Hashes is enabled and the logger will be the one from the cache.

This will re-use previous results that we save in the cachingLogger.